src/signTx.c

```diff
@@ -687,11 +721,7 @@ static void signTx_handleInputAPDU(uint8_t p2, uint8_t* wireDataBuffer, size_t w
 		ASSERT(wireDataSize < BUFFER_SIZE_PARANOIA);
 	}
 
-	// parsed data
-	struct {
-		uint8_t txHashBuffer[TX_HASH_LENGTH];
-		uint32_t parsedIndex;
-	} input;
+	sign_tx_input_data_t* input = &txBodyCtx->stageData.input;
```

Comment:
You should have a look at #108, because these modifications will clump up into a backlog.